### DISCUSSION TITLE
After zooming in too much, straight line shifts upwards

### DISCUSSION THREAD

**Discussion by ale-coe** [chartjs/chartjs-plugin-zoom#874] [Q&A]
### Current behavior
In an unzoomed state the blue dot is located directly on the line (the line itself is an annotation and the blue dot is part of a scatter chart).
![img_1](https://github.com/chartjs/chartjs-plugin-zoom/assets/24796354/1fb4f757-4c49-42f0-80da-483d4cb90213)
If I zoom in too much, the line shifts upwards and the dot is not located on the line anymore.
![img_2](https://github.com/chartjs/chartjs-plugin-zoom/assets/24796354/80513bcc-8667-4490-8646-62d86e7cc7b5)

Swapping the annotation with a line chart or a scatter chart does not fix the issue.

### Expected behavior
The zoomlevel does not affect how the line is displayed.

### Reproducible sample
See this [codepen](https://codepen.io/Yawgmoth1806/pen/yLZPbGj).

**Comment by fehmisener** [comment]
Hey,

I'm having the same problem, did you find a solution?

**Comment by kurkle** [comment]
This is (at least) related to the fact that getPixelForValue limites its result to -32767..32767. When the annotation (or line) closest point is far enough in some direction, it gets limited. If I remember correctly, canvas has similar limitations.

maybe limit the zooming? and/or use a line chart instead of annotation with many points.
